compiler-rt/lib/asan/CMakeLists.txt

```diff
@@ -1,5 +1,12 @@
 # Build for the AddressSanitizer runtime support library.
 
+set(COMPILER_RT_ASAN_BUILD_SHARED_LIBS_default ON)
+if (CMAKE_SYSTEM_NAME STREQUAL "WASI")
```

Comment:
@sbc100 I assume we've never tried the ASan CMake build for emscripten since we have the standalone build scripts?
Anyway this might be a case where we want the same logic for WASI and Emscripten.